joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/frontend/TypeNodePass.scala

```diff
@@ -74,6 +74,11 @@ object TypeNodePass {
   }
 
   def fullToShortName(typeName: String): String = {
-    typeName.takeWhile(_ != ':').split('.').lastOption.getOrElse(typeName)
+    if (typeName.endsWith(">")) {
```

Comment:
Example:
Without this change: `"foo.bar<package.someGeneric>" -> "someGeneric>"`
With this change: `"foo.bar<package.someGeneric>" -> "bar"`